### DISCUSSION TITLE
What feature creates this strange url?

### DISCUSSION THREAD

**Discussion by AliceDTRH** [community/community#32478] [Apps, API and Webhooks] [answered]
What feature creates url's like this: https://github.com/ryanoasis/nerd-fonts/archive/v2.2.1.tar.gz
And how might I get additional information about the uploaded file (e.g. release date, sha hashes, etc)?

Kind regards,
Alice

**Comment by airtower-luna** [comment]
It's the source archive download you can get for releases and with the "download ZIP" feature: repository URL + `archive/` + the commit/ref you want + file type ending

Note that if you go through the web UI the ref includes the full ref type, e.g. `refs/tags/v2.2.1`, the form you posted is ambiguous: It could refer to a branch or tag of the name, but I assume `v2.2.1` will be a tag. So you should be able to find `v2.2.1` on the releases page for the repository, or by retrieving a tag of the name.

**Reply by AliceDTRH** [reply]
Thank you, your answer helped as well!

**Comment by jfullstackdev** [comment, marked as answer]
> What feature creates url's like this:

you are referring to releases, if your repo have releases, you will be provided with those options when you create the release, simply right click it and you will have the link, like this

https://github.com/jdevstatic/lightweight-web-server/archive/refs/tags/v3.2.tar.gz

![image](https://user-images.githubusercontent.com/47092464/189491135-1dbe7e84-3684-4f2a-a28c-b2ddf5d55437.png)

![image](https://user-images.githubusercontent.com/47092464/189491151-fe80fa16-b6c4-4c32-a9d6-49640cf03f99.png)

this is the link to the repo 

https://github.com/jdevstatic/lightweight-web-server/releases

**Reply by AliceDTRH** [reply]
Oooooh, it's the automatically generated source code zips! Thank you :D
